com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/validation/OpenApi3Validator.java

```diff
@@ -42,19 +43,74 @@ private void validateSecuritySchemeReferences(Model model, AbstractNode node, Se
 
             if (node.isObject()) {
                 for (String field : node.asObject().fieldNames()) {
-                    if (securitySchemes.get(field) == null) {
+                    AbstractNode securityScheme = securitySchemes.get(field);
+
+                    if (securityScheme == null) {
                         errors.add(
                                 error(node.get(field), IMarker.SEVERITY_ERROR, Messages.error_invalid_reference_type));
+                    } else {
+                        validateSecuritySchemScopes(node, field, securityScheme, errors);
                     }
                 }
             }
         }
     }
 
+    private List<String> oauthScopes = Lists.newArrayList("oauth2", "openIdConnect");
+
+    private void validateSecuritySchemScopes(AbstractNode node, String name, AbstractNode securityScheme,
+            Set<SwaggerError> errors) {
+        AbstractNode type = securityScheme.get("type");
+        if (type == null) {
+            return;
+        }
+
+        boolean hasScopes = oauthScopes.contains(type.asValue().getValue());
```

Comment:
`hasScopes` sounds like the node already has defined scope, while it's more like `shouldHaveScopes`